app/src/main/java/org/mozilla/focus/locale/screen/LocaleFragment.kt

```diff
@@ -0,0 +1,149 @@
+import org.mozilla.focus.ui.theme.FocusTheme
+import java.util.Locale
+
+class LocaleFragment : BaseSettingsLikeFragment() {
+    private val viewModel: LocaleViewModel by viewModels()
+    private var _binding: FragmentLocaleBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentLocaleBinding.inflate(inflater, container, false)
+        setupViewModel()
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        updateTitle(R.string.preference_language)
+    }
+
+    private fun setupViewModel() {
+        viewModel.getAllLocaleNameAndTag(
+            requireActivity().getString(R.string.preference_language_systemdefault),
+            getCurrentLocale()
+        )
+
+        viewModel.loadingLiveData.observe(
+            viewLifecycleOwner,
+            {
+                if (it) {
+                    binding.localeLoading.visibility = View.VISIBLE
+                } else {
+                    binding.localeLoading.visibility = View.GONE
+                }
+            }
+        )
+
+        viewModel.languagesLiveData.observe(
+            viewLifecycleOwner,
+            {
+                binding.aboutLocaleContent.setContent {
+                    val listState = rememberLazyListState()
+                    val coroutineScope = rememberCoroutineScope()
+                    coroutineScope.launch {
+                        listState.scrollToItem(viewModel.selectedIndex)
+                    }
+                    val state = remember {
+                        mutableStateOf(getCurrentLocale())
+                    }
+                    FocusTheme {
+                        LanguagesList(
+                            languageList = createLanguageListItem(it, state = state),
+                            state = state,
+                            listState = listState
+                        )
+                    }
+                }
+            }
+        )
+    }
+
+    private fun createLanguageListItem(
+        languages: List<Language>,
+        state: MutableState<String>
+    ): List<LanguageListItem> {
+        val languageListItems = ArrayList<LanguageListItem>()
+        languages.forEach { language ->
+            val languageListItem = LanguageListItem(
+                language = language,
+                onClick = {
+                    state.value = language.tag
+                    saveCurrentLocaleInSharePref(language.tag)
+                    setCurrentLanguage(language.tag)
+                }
+            )
+            languageListItems.add(languageListItem)
+        }
+        return languageListItems
+    }
+
+    private fun getCurrentLocale(): String {
```

Comment:
If we are using a ViewModel to handle the business logic then the responsibility of deciding where/how to save the current locale should be extracted from the Fragment.